libsolidity/codegen/ir/IRLValue.cpp

```diff
@@ -166,3 +166,79 @@ string IRStorageArrayLength::setToZero() const
 {
 	return storeValue("0", *TypeProvider::uint256());
 }
+
+IRMemoryItem::IRMemoryItem(
+	IRGenerationContext& _context,
+	std::string _address,
+	bool _byteArrayElement,
+	Type const& _type
+):
+	IRLValue(_context, &_type),
+	m_address(move(_address)),
+	m_byteArrayElement(_byteArrayElement)
+{ }
+
+string IRMemoryItem::retrieveValue() const
+{
+	if (m_byteArrayElement)
+		return "byte(0, mload(" + m_address + ")";
```

Comment:
```suggestion
		return "byte(0, mload(" + m_address + "))";
```

Reply:
Actually, I think the `byte(0` call here is wrong. It would be `mload` plus cleanup.